db-tests/src/test/java/TestTriggers.java

```diff
@@ -82,14 +90,491 @@ void afterall() throws SQLException, IOException, InterruptedException {
+                        .formatted(fileId)
+                );
+
+      final var updatedRes = connection.createStatement()
+                                       .executeQuery(
+                                           """
+                                               SELECT revision
+                                               FROM mission_model
+                                               WHERE id = %s;"""
+                                               .formatted(missionModelId)
+                                       );
+      updatedRes.next();
+      final var updatedRevision = updatedRes.getInt("revision");
+      updatedRes.close();
+
+      assertEquals(revision + 1, updatedRevision);
+    }
+  }
+
+  @Nested
+  class PlanTriggers {
+    @Test
+    void shouldIncrementPlanRevisionOnPlanUpdate() throws SQLException {
+      final var initialRes = connection.createStatement()
+                                       .executeQuery(
+                                           """
+                                               SELECT revision FROM plan
+                                               WHERE id = %s;"""
+                                               .formatted(planId)
+                                       );
+      initialRes.next();
+      final var initialRevision = initialRes.getInt("revision");
+      initialRes.close();
+
+      connection.createStatement()
+                .executeUpdate(
+                    """
+                        UPDATE plan SET name = 'test-plan-updated-%s'
+                        WHERE id = %s;"""
+                        .formatted(UUID.randomUUID().toString(), planId)
+                );
+
+      final var updatedRes = connection.createStatement()
+                                       .executeQuery(
+                                           """
+                                               SELECT revision FROM plan
+                                               WHERE id = %s;"""
+                                               .formatted(planId)
+                                       );
+      updatedRes.next();
+      final var updatedRevision = updatedRes.getInt("revision");
+      updatedRes.close();
+
+      assertEquals(initialRevision + 1, updatedRevision);
+    }
+
+    @Test
+    void shouldIncrementPlanRevisionOnActivityInsert() throws SQLException {
+      final var initialRes = connection.createStatement()
+                                       .executeQuery(
+                                           """
+                                               SELECT revision FROM plan
+                                               WHERE id = %s;"""
+                                               .formatted(planId)
+                                       );
+      initialRes.next();
+      final var initialRevision = initialRes.getInt("revision");
+      initialRes.close();
+
+      final var _activityId = insertActivity(planId);
```

Comment:
I don't think this variable is used in this test so you can remove it.

Reply:
I prefixed it with the underscore to indicate that its unused. I wanted to ensure that the behavior of insertActivity was still clear (that it returns the activity id). Perhaps this is a holdover from typescript style and shouldn't be applied in Java

Reply:
ahh, I have seen that with other languages, One language the` _ ` on variables or method means it is private. Got to love all them different programming languages.

If you want to verify `insertActivity(planid)` returns an activity id (int) you can do a assertTrue:

`assertTrue(Integer.class.isInstance(insertActivity(planId)));`

Reply:
Removed the assigment